src/tuning.ts

```diff
@@ -19,59 +24,63 @@ export class Tuning {
 
   constructor(tuningString = 'E/5,B/4,G/4,D/4,A/3,E/3,B/2,E/2') {
     // Default to standard tuning.
+    this.tuningString = '';
+    this.tuningValues = [];
+    this.numStrings = 0;
     this.setTuning(tuningString);
   }
 
-  noteToInteger(noteString) {
+  static noteToInteger(noteString: string): number {
     return Flow.keyProperties(noteString).int_value;
   }
 
-  setTuning(noteString) {
+  setTuning(noteString: string): void {
+    let noteName = noteString;
     if (Tuning.names[noteString]) {
-      noteString = Tuning.names[noteString];
+      noteName = Tuning.names[noteString];
     }
 
-    this.tuningString = noteString;
+    this.tuningString = noteName;
     this.tuningValues = [];
     this.numStrings = 0;
 
-    const keys = noteString.split(/\s*,\s*/);
+    const keys = noteName.split(/\s*,\s*/);
     if (keys.length === 0) {
-      throw new Vex.RERR('BadArguments', 'Invalid tuning string: ' + noteString);
+      throw new Vex.RERR('BadArguments', `Invalid tuning string: ${noteString}`);
     }
 
     this.numStrings = keys.length;
-    for (let i = 0; i < this.numStrings; ++i) {
-      this.tuningValues[i] = this.noteToInteger(keys[i]);
+    for (let i = 0; i < this.numStrings; i += 1) {
+      this.tuningValues[i] = Tuning.noteToInteger(keys[i]);
     }
   }
 
-  getValueForString(stringNum) {
-    const s = parseInt(stringNum, 10);
+  getValueForString(stringNum: string | number): number {
+    const s = typeof stringNum === 'string' ? parseInt(stringNum, 10) : stringNum;
```

Comment:
checked that `parseInt()` on possible int only occurs about 15 times in the code, so no need for a helper function, it won't be everywhere.

Reply:
@mscuthbert @0xfe this is an interface change. Should we do it?

Reply:
I agree that this is used with string nowhere. Probably the best is to change the parameter to number. All those checks also consume processing time.

Reply:
Sorry, I didn't mean to remove "| number" from parameter type.

Reply:
I just was wondering if we needed something like this to be written once so that we don't need to rewrite it every time it appears in the code:

```
export function possibleNumberAsString(stringOrNum: string | number): number {
    return typeof stringNum === 'string' ? parseInt(stringNum, 10) : stringNum;
}

const s = possibleNumberAsString(stringNum);
```

but the particular `parseInt(stringNum, 10)` code only appears about 12 times in the entire codebase, so we wouldn't need it too often.  (but it would make it clearer what the `const s =` declaration is doing.

Reply:
Pro-tip: `const s = Number(stringNum)` 🥇